### PR TITLE
New unit test to scan for external links/forms and if they have rel="noreferrer noopener"

### PR DESCRIPTION
**Pull request by woutersamaey**
@NicolasDorier this is the continuation of previous PR #2655 
- New unit test
- Fixed all templates
- URLs that are inside a variable cannot be identified as internal or external. So, I added "noreferrer noopener" to them anyway as it does no harm. This is like 1-2% of cases. Should not be an issue.
- Renamed the original method `CheckLinks` to `CheckDeadLinks` since this test is also checking links and to avoid confusion.

### CODE REVIEW

File: BTCPayServer.Tests/UnitTest1.cs [hunk 1]
```diff
@@ -206,12 +206,79 @@ public async Task CheckNoDeadLink()
...
+            foreach (var file in viewFiles)
+            {
+                var html = await File.ReadAllTextAsync(file);
+
+                checkHtmlNodesForReferrer(file, html, "a", "href");
+                checkHtmlNodesForReferrer(file, html, "form", "action");
+            }
+        }
+
+        private void checkHtmlNodesForReferrer(string filePath, string html, string tagName, string attribute)
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:232]: nit: PascalCase
```suggestion
        private void CheckHtmlNodesForReferrer(string filePath, string html, string tagName, string attribute)
```

File: BTCPayServer.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -206,12 +206,79 @@ public async Task CheckNoDeadLink()
...
+                            var relWords = rel.Split(" ");
+                            Assert.Contains("noreferrer", relWords);
+                            Assert.Contains("noopener", relWords);
+                        }
+                    }
+                }
+            }
+        }
+
+        private String getAttributeValue(String nodeHtml, string attribute)
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:270]: nit:PascalCase
```suggestion
        private String GetAttributeValue(String nodeHtml, string attribute)
```

**Review by Kukks — APPROVED** [review]

**Comment by woutersamaey** [comment]
Thanks for your approval @Kukks

**Review by Kukks — CHANGES_REQUESTED** [review]
Seems like there may be a malformed HTML tag in this or in Notifications/Index

**Comment by woutersamaey** [comment]
Possibly a bad merge. WIll check.

**Comment by Kukks** [comment]
(you'll also need to update the calls with the new pascal casing)

**Comment by woutersamaey** [comment]
@Kukks should be OK now.

**Pull request merged**